geopandas/tools/overlay.py

```diff
@@ -373,5 +373,6 @@ def overlay(df1, df2, how='intersection', make_valid=True, use_sindex=None):
         dfunion = _overlay_union(df1, df2)
         result = dfunion[dfunion['__idx1'].notnull()].copy()
         result.reset_index(drop=True, inplace=True)
-    result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
+    if set(result.columns).intersection(['__idx1', '__idx2'])!=set():
```

Comment:
I would maybe simply check if one of both is in the columns, that should be sufficient (`if '__idx1' in result.columns:`)

Reply:
Or alternatively, use `errors='ignore'` in the `drop` call, or add `['__idx1', '__idx2']` to the columns when creating the empty dataframe inside `_overlay_intersection`, so we don't have to special case here.

Reply:
Perhaps adding the columns in the empty dataset may be the best. This ensures we only do the operation if needed.

Reply:
For the test I need to create a third polygon. I can either add it only in the test or do you want me to add it in where ``df1`` and ``df2`` are created? That would require to change a whole bunch of code.

Reply:
I would just add it in the test. Currently many of the tests reuse the same data, but for very specific tests it is totally fine to just create the two dataframes you need in the test itself.